src/components/vm/filesystems/vmFilesystemsCard.jsx

```diff
@@ -0,0 +1,254 @@
+                      gridBreakPoint='grid-xl'
+                      emptyCaption={_("No directories shared between the host and this VM")}
+                      columns={columnTitles}
+                      rows={rows} />
+    );
+};
+
+export const VmFilesystemActions = ({ connectionName, dispatch, objPath, vmName, vmState, memory, memoryBacking }) => {
+    const [additionalOptionsExpanded, setAdditionalOptionsExpanded] = useState(false);
+    const [dialogError, setDialogError] = useState();
+    const [isOpen, setIsOpen] = useState(false);
+    const [memoryBackingType, setMemoryBackingType] = useState("file");
+    const [mountTag, setMountTag] = useState("");
+    const [source, setSource] = useState("");
+    const [validationFailed, setValidationFailed] = useState({});
+    const [xattr, setXattr] = useState(false);
+    const idPrefix = `${vmId(vmName)}-filesystems`;
+
+    const onAddClicked = () => {
+        const validationFailed = {};
+
+        if (!mountTag)
+            validationFailed.mountTag = _("Mount tag must not be empty");
+        if (!source)
+            validationFailed.source = _("Source must not be empty");
+
+        setValidationFailed(validationFailed);
+
+        if (Object.getOwnPropertyNames(validationFailed).length == 0) {
+            setMemoryBacking({
+                connectionName, objPath,
+                type: memoryBackingType,
+                memory
+            })
+                    .then(() => {
+                        createFilesystem({
+                            connectionName, objPath,
+                            source, target: mountTag,
+                            xattr,
+                        });
+                    })
+                    .then(
+                        () => {
+                            setAdditionalOptionsExpanded(false);
+                            setDialogError();
+                            setIsOpen(false);
+                            setMemoryBackingType('file');
+                            setMountTag("");
+                            setSource("");
+                            setValidationFailed({});
+                            setXattr(false);
+                        },
+                        exc => setDialogError(exc.message)
+                    );
+        }
+    };
+    const addButton = (
+        <Button id={`${idPrefix}-add`}
+                isAriaDisabled={vmState != 'shut off'}
+                onClick={() => setIsOpen(true)}
+                variant="secondary">
+            {_("Add shared directory")}
+        </Button>
+    );
+
+    return (
+        <>
+            {vmState == 'shut off' ? addButton : <Tooltip content={_("Adding shared directories is possible only when the guest is shut off")}>{addButton}</Tooltip>}
+            <Modal position="top" isOpen={isOpen} variant="medium" onClose={() => setIsOpen(false)}
+                   title={_("Add a shared host directory with the guest")}
```

Comment:
Maybe 'Share a host directory with the guest', sounds better.

Reply:
I agree. "Share a host directory with the guest" does sound better.

Does that mean we should update the action buttons too?

"Share host directory" / "Share directory"

Reply:
Sounds indeed better than 'add'. Will update! Thanks